### PR TITLE
add binary cross entropy with logit loss

### PR DESCRIPTION
**Pull request by ZHUI**
### PR types

New features
### PR changes

APIs
### Describe

add binary cross entropy with logit loss

### CODE REVIEW

File: python/paddle/nn/functional/loss.py [hunk 1]
```diff
@@ -73,6 +75,154 @@
...
+    weight on the positive label of each class.
+
+    Finally, this operator applies reduce operation on the loss.
+    If :attr:`reduction` set to ``'none'``, the operator will return the original loss `Out`.
+    If :attr:`reduction` set to ``'mean'``, the reduced mean loss is $$ Out = MEAN(Out) $$.
+    If :attr:`reduction` set to ``'sum'``, the reduced sum loss is $$ Out = SUM(Out) $$.
+
+    Note that the target labels ``label`` should be numbers between 0 and 1.
+
+    Parameters:
```
Comment (wawltor) [inline python/paddle/nn/functional/loss.py]: parameters->Args

File: python/paddle/nn/functional/loss.py [hunk 2]
```diff
@@ -73,6 +75,154 @@
...
+        logit (Tensor): The input predications tensor. 2-D tensor with shape: [N, *],
+            N is batch_size, `*` means number of additional dimensions. The ``logit``
+            is usually the output of Linear layer. Available dtype is float32, float64.
+        label (Tensor): The target labels tensor. 2-D tensor with the same shape as
+            ``input``. The target labels which values should be numbers between 0 and 1.
```
Comment (wawltor) [inline python/paddle/nn/functional/loss.py]: input->logit

File: python/paddle/nn/functional/loss.py [hunk 3]
```diff
@@ -73,6 +75,154 @@
...
+        name (str, optional): Name for the operation (optional, default is None).
+            For more information, please refer to :ref:`api_guide_Name`.
+
+    Returns:
+        output (Tensor): If ``reduction`` is ``'none'``, the shape of output is
+            same as ``input`` , else the shape of output is scalar.
+
+    Examples:
+
+        .. code-block:: python
```
Comment (wawltor) [inline python/paddle/nn/functional/loss.py:291]: 空一行

File: python/paddle/nn/functional/loss.py [hunk 4]
```diff
@@ -73,6 +75,154 @@
...
+            import paddle
+            paddle.disable_static()
+            input = paddle.to_tensor([5.0, 1.0, 3.0], dtype="float32")
+            label = paddle.to_tensor([1.0, 0.0, 1.0], dtype="float32")
+            output = paddle.nn.functional.binary_cross_entropy_with_logits(input, label)
+            print(output.numpy())  # [0.45618808]
+            paddle.enable_static()
```
Comment (wawltor) [inline python/paddle/nn/functional/loss.py]: delete 一行

File: python/paddle/nn/layer/loss.py [hunk 5]
```diff
@@ -32,6 +33,108 @@
...
+            Default is ``'None'``.
+        name (str, optional): Name for the operation (optional, default is None).
+            For more information, please refer to :ref:`api_guide_Name`.
+
+    Shapes:
+        logit (Tensor): The input predications tensor. 2-D tensor with shape: [N, *],
+            N is batch_size, `*` means number of additional dimensions. The ``logit``
+            is usually the output of Linear layer. Available dtype is float32, float64.
+        label (Tensor): The target labels tensor. 2-D tensor with the same shape as
+            ``input``. The target labels which values should be numbers between 0 and 1.
```
Comment (wawltor) [inline python/paddle/nn/layer/loss.py]: input->logit

File: python/paddle/nn/layer/loss.py [hunk 6]
```diff
@@ -32,6 +33,108 @@
...
+            Available dtype is float32, float64.
+        output (Tensor): If ``reduction`` is ``'none'``, the shape of output is
+            same as ``input`` , else the shape of output is scalar.
+
```
Comment (wawltor) [inline python/paddle/nn/layer/loss.py:105]: 加个Return 保持统一

File: python/paddle/nn/functional/loss.py (hunk 1 above)
Comment (ZHUI) [reply python/paddle/nn/functional/loss.py]: done

File: python/paddle/nn/functional/loss.py (hunk 2 above)
Comment (ZHUI) [reply python/paddle/nn/functional/loss.py]: done

File: python/paddle/nn/functional/loss.py (hunk 4 above)
Comment (ZHUI) [reply python/paddle/nn/functional/loss.py]: done

File: python/paddle/nn/functional/loss.py (hunk 3 above)
Comment (ZHUI) [reply python/paddle/nn/functional/loss.py:291]: done

File: python/paddle/nn/layer/loss.py (hunk 6 above)
Comment (ZHUI) [reply python/paddle/nn/layer/loss.py:105]: done

File: python/paddle/nn/layer/loss.py (hunk 5 above)
Comment (ZHUI) [reply python/paddle/nn/layer/loss.py]: done

**Review by wawltor — APPROVED** [review]
LGTM
